### DISCUSSION TITLE
Test Account/Bypass 2FA for GitHub OAuth

### DISCUSSION THREAD

**Discussion by sanskar10100** [community/community#31459] [Other Feature Feedback, Questions, & Ideas] [answered]
I built an Android app for developers where they can share code snippets. The app requires that a developer sign in with GitHub before they continue. We're using GitHub OAuth for this, with the web flow where the app launches the login page in a browser, and then the user is redirected to the app after a successful login.

I already publish the app through GitHub Releases, but for a wider reach, it submitted this app to the Play Store, but since login is mandatory for them to test the app I had to give them credentials. I submitted a test GitHub account that I had lying around, but now the issue is that it requires an OTP due to 2FA, which results in our app getting rejected during the review. According to them, we didn't supply the following:

> Reusable login credentials that do not expire or require 2-step verification

Is there a test account that we can use? Or perhaps a way to bypass the 2FA for a specific account?

**Comment by joshuawalker** [comment, marked as answer]
Hey! 👋 

It's not possible to bypass 2FA when it is enabled, or disable the requirement of non-2FA enabled accounts to verify new devices when signing in from an unrecognized location.

Your best bet would be to enable two-factor authentication and provide the authentication secret to the reviewers along with the account password. There may be third-party services that can assist with securely sharing this credential. Providing a recovery code wouldn't work because using a recovery code interrupts the OAuth flow.
